js/web/lib/wasm/jsep/webgpu/ops/binary-op.ts

```diff
@@ -172,16 +153,31 @@ const createBinaryOpProgramInfo =
         // element-wise
         vectorize = true;
       }
-
+      cacheKey += vectorize;
+      const useShapesUniforms = isBroadcast && enableShapesUniforms(a.dims.length) &&
```

Comment:
`isBroadcast` should be not related with `useShapesUniforms`. Or you can specify each input/output whether they want to use uniform.
```
const useShapesUniforms = enableShapesUniforms(a.dims.length) &&
          enableShapesUniforms(b.dims.length) && enableShapesUniforms(outputShape.length);

Reply:
We want to not set useShapeUniform when we vectorize. May be we could use vectorize instead of !isBroadcast here

Reply:
Keeping it simple by not using isBroadcast or !vectorize trying to avoid creating unnecessary/unused shape and strides variables.